### ISSUE TITLE
SharePoint PnP Search v4 - Not able to search for custom managed properties

### ISSUE THREAD

**Issue by sonyszkiewicz** [microsoft-search/pnp-modern-search#2900]
Hello,

I have setup people search based on PnPsearch v4. One of the requirement is to allow users look for employees using their trigram (3 letters). 

Steps to reproduce:

Create new user properties in SharePoint Admin Center
update the property with approprieate data
Map crawled property to managed property (RefinableStringXX)
Configured PnPSearch

Once i configured PnPSearch i should be able to search users using the trigram, however I am not getting any resoults.

The new property is available as a refiner in PnP Search filter.

![2023-04-11 18_27_40-People Finder and 9 more pages - Work 2 - Microsoft​ Edge](https://user-images.githubusercontent.com/98872520/231228591-84db726e-33c6-44c0-a393-079dd2fd9ee3.png)
![2023-04-11 18_26_47-Edit User Profile Property and 9 more pages - Work 2 - Microsoft​ Edge](https://user-images.githubusercontent.com/98872520/231228602-792954bd-2ca4-48ee-a4bc-107383813877.png)
![2023-04-11 18_29_16-People Finder and 9 more pages - Work 2 - Microsoft​ Edge](https://user-images.githubusercontent.com/98872520/231228606-9180c7c3-7a9f-4fbe-9325-0ac96883979e.png)

**Comment by kasperbolarsen** [comment]
I am not at my pc right now, but I am pretty sure that you will need create a new managed property and map it to the crawled property ( remember to select people peopleidx under advanced settings) in order to search for the initials.

**Comment by kasperbolarsen** [comment]
Yes, that is correct.  The reason is, AFAIK that the Refinablestring isn't queryable.

**Comment by sonyszkiewicz** [comment]
unfortunately no changes

Properties are configured as on the screensow belo


![2023-04-12 13_15_25-View Managed Property - RefinableString13 and 9 more pages - Work - Microsoft​ E](https://user-images.githubusercontent.com/98872520/231441490-70b5cf69-e1c2-47c2-b76f-a49e7da55347.png)
![2023-04-12 13_13_46-View Managed Property - Trigram and 9 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/98872520/231441497-5ef19a1b-c36a-45b6-9a2d-00cf496fcb7c.png)



![2023-04-12 13_13_22-View Crawled Property - urn_schemas-microsoft-com_sharepoint_portal_profile_Trig](https://user-images.githubusercontent.com/98872520/231441486-1a472a36-7867-4da7-aad2-ad16536c2f0d.png)

**Comment by bridgestateu** [comment]
I am following this thread as I too have created a custom SharePoint user property and I have mapped the corresponding Search crawled property to a managed property of type RefinableStringXX. I am unable to query this new managed property in the PNP Modern Search web parts using the alias or the RefinableStringXX name. I am finding too that I cannot query this new managed property in the Microsoft Graph Explorer either.

**Comment by kasperbolarsen** [comment]
@bridgestateu Are you using the SharePoint or Microsoft datasource in your results web part? If you are using SharePoint the the data is coming from the SharePoint search index, specifically the peopleindex.
You should be able to verify that data in the index using either Sp editor or SharePoint search query tool.

**Comment by bridgestateu** [comment]
@kasperbolarsen I am using the SharePoint data source inside the results web part. After reviewing this thread today, I have modified the managed property's advanced searchable settings to PeopleIdx. I have not yet been able to return this managed property in the web part nor in the Graph Explorer. I am thinking that I need to wait until the user profiles are reindexed. I assume this could take several hours and I will hope to see a change tomorrow.

**Comment by kasperbolarsen** [comment]
I always add both the Refinablestring and the new Managed property  ( both mapped to the same crawled property) in order to be able to fulltext search using the query and to refine the results using the Refinablestring.

**Comment by bridgestateu** [comment]
Acknowledged and I see that conversation above. You have advised that the RefinableStringXX is not searchable (and possibly why I am not seeing this when I select it as a property). So, I should use a new custom managed property in order to select that property and use the RefinableStringXX property in the configuration of the filter web part.

**Comment by kasperbolarsen** [comment]
And remember that the new managed property must be set to use the peopleindex under advanced settings (only the case when the property is a custom user profile property)

**Comment by bridgestateu** [comment]
@kasperbolarsen thank you so much for this advice. I did use the peopleindex under advanced settings.

<img width="878" alt="image" src="https://user-images.githubusercontent.com/77447096/231866773-4e1f46d3-14c3-4000-b703-1969f50a7a92.png">

**Comment by bridgestateu** [comment]
@kasperbolarsen with reference to the above, I understand now that I should select the 'BSUDivisionName' property in the PNP results web part and use the RefinableString01 by its alias name 'Division' in the filters web part.

**Comment by kasperbolarsen** [comment]
I guess so, I haven't tried to use the aliases. Both the BSUDivisionName property and the Refinablestring must be in selected properties for it to work.

**Comment by sonyszkiewicz** [comment]
That might be a silly question, but how do i run this  script https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps

i am following the guide provided but nothing is happening, not being prompted for sign in or anything

**Comment by bridgestateu** [comment]
@sonyszkiewicz  I have started this script this morning. First, have you been able to successfully load the module? I had to modify the import command. I created the psm1 file and changed to the directory in PS. I got an error with the documented command because PS couldn't resolve the path. So I typed:

**Import-Module .\reindexusers.psm1 -Verbose**

and it loaded the module. Then I typed this command with our own domain name:

**Request-PnPReindexUserProfile -url https://contoso.sharepoint.com**

It ran and loaded all of the user profiles ...

![image](https://user-images.githubusercontent.com/77447096/232049381-00fb587e-c615-4c93-bba3-61f2b635e2f3.png)

**Comment by kasperbolarsen** [comment]
> That might be a silly question, but how do i run this  script https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps
> 
> i am following the guide provided but nothing is happening, not being prompted for sign in or anything

If you have any issues using the sample please raise an issue in the samples github, in order to let the authors know

**Comment by wobba** [comment]
Easiest to make a people property searchable is to also map it to the managed property `ContentsHidden`. Mentioned at the bottom of my post https://www.techmikael.com/2014/11/how-to-make-sure-user-profile.html.

And a re-index has to be completed after the schema mapping to take effect as mentioned above.
